runtime/resolvers/sql.go

```diff
@@ -115,6 +117,10 @@ func (r *sqlResolver) ResolveInteractive(ctx context.Context) (*runtime.Resolver
 
 	var out []map[string]any
 	for res.Rows.Next() {
+		if len(out) >= sqlResolverInteractiveRowLimit {
```

Comment:
this may not protect backend but only runtime as backends may materialize full results before returning results, whats the purpose ? for backend should we add limit clause but it can be tricky

Reply:
I updated the PR to wrap the SQL in an outer select that pushes the limit into the DB. Might not be perfect for all edge cases, but seems pretty good for now.